extensions/amp-gwd-animation/0.1/amp-gwd-animation-impl.js

```diff
@@ -0,0 +1,447 @@
+  setCurrentPage(index) {
+    // Turn off animations on the previously-active page, if there was one.
+    // TODO(sklobovskaya): Decide if it's worth just storing the index.
+    const currentPageEl = this.ampdoc_.getRootNode().querySelector(
+        `.${GWD_PAGE_WRAPPER_CLASS}.${PlaybackCssClass.PLAY}`);
+
+    if (currentPageEl) {
+      currentPageEl.classList.remove(PlaybackCssClass.PLAY);
+    }
+
+    // Activate animations on the new current page.
+    const gwdPages = this.ampdoc_.getRootNode().querySelectorAll(
+        `.${GWD_PAGE_WRAPPER_CLASS}`);
+    const newPageEl = gwdPages[index];
+
+    if (newPageEl) {
+      newPageEl.classList.add(PlaybackCssClass.PLAY);
+    }
+  }
+
+  /**
+   * The play action.
+   * @param {string} id Receiver id.
+   */
+  play(id) {
+    const receiver = this.getReceiver(id);
+
+    if (!receiver) {
+      return;
+    }
+
+    receiver.classList.remove(PlaybackCssClass.PAUSE);
+  }
+
+  /**
+   * The pause action.
+   * @param {string} id Receiver id.
+   */
+  pause(id) {
+    const receiver = this.getReceiver(id);
+
+    if (!receiver) {
+      return;
+    }
+
+    receiver.classList.add(PlaybackCssClass.PAUSE);
+  }
+
+  /**
+   * The togglePlay action.
+   * @param {string} id Receiver id.
+   */
+  togglePlay(id) {
+    const receiver = this.getReceiver(id);
+
+    if (!receiver) {
+      return;
+    }
+
+    receiver.classList.toggle(PlaybackCssClass.PAUSE);
+  }
+
+  /**
+   * The gotoAndPlay action.
+   * @param {string} id Receiver id.
+   * @param {string} label The name of the label animation to go to.
+   */
+  gotoAndPlay(id, label) {
+    const receiver = this.getReceiver(id);
+
+    if (!receiver) {
+      return;
+    }
+
+    this.playLabelAnimation_(receiver, label);
+  }
+
+  /**
+   * The gotoAndPause action. A gotoAndPause is a gotoAndPlay followed by an
+   * (almost) immediate pause.
+   * @param {string} id Receiver id.
+   * @param {string} label The name of the label animation to go to.
+   */
+  gotoAndPause(id, label) {
+    const receiver = this.getReceiver(id);
+
+    if (!receiver) {
+      return;
+    }
+
+    // Switch to the label animation.
+    this.playLabelAnimation_(receiver, label);
+
+    // Trigger a pause.
+    const timeToWaitBeforePause = 40;
+
+    this.ampdoc_.win.setTimeout(() => {
+      this.pause(id);
+    }, timeToWaitBeforePause);
+  }
+
+  /**
+   * The gotoAndPlayNTimes action.
+   * @param {string} id Receiver id.
+   * @param {string} label The name of the label animation to go to.
+   * @param {number} count The number of times to repeat this gotoAndPlay.
+   * @param {string} eventName The timeline event name, used to record an
+   *     counter for this gotoAndPlayNTimes invocation.
+   */
+  gotoAndPlayNTimes(id, label, count, eventName) {
+    if (count <= 0) {
+      user().warn(LOG_ID, `Invalid count parameter ${count}.`);
+      return;
+    }
+
+    if (!eventName) {
+      user().warn(LOG_ID, `Invalid event name ${eventName}.`);
```

Comment:
Why warning? Should we just do an error?

Reply:
Yes, these can just be errors. Done.